### ISSUE TITLE
Setting TestCaseAttribute.TestName in .NET Core project causes not running tests

### ISSUE THREAD

**Issue by tomaszskrobek** [nunit/nunit#3522]
Named tests are not running via Visual Studio Test Explorer when using .NET Core. 

![image](https://user-images.githubusercontent.com/63336798/78754016-10507980-7977-11ea-8dba-18f8fe05eea0.png)

Logs from Test Output:
`[08.04.2020 8:59:20.647 AM Diagnostic] Enqueue operation 'RunFromContextOperation', hashcode:46380802 
[08.04.2020 8:59:20.648 AM Diagnostic] Operation left in the the queue: 1
[08.04.2020 8:59:20.648 AM Diagnostic] 	'RunFromContextOperation', hashcode:46380802
[08.04.2020 8:59:20.648 AM Diagnostic] 

[08.04.2020 8:59:20.648 AM Diagnostic] Operation Dequeue : 'RunFromContextOperation'
[08.04.2020 8:59:20.693 AM Diagnostic] Starting programmatic build of containers... 
[08.04.2020 8:59:20.807 AM Diagnostic] Completed programmatic build of containers.
[08.04.2020 8:59:20.807 AM Diagnostic] TestContainer update (build) complete : 114 ms
[08.04.2020 8:59:20.808 AM Diagnostic] DiscoveryOperation<RunFromContextOperation> FinishedChangedCotainers, changed container count is 1
[08.04.2020 8:59:20.808 AM Diagnostic] Discovering the following containers :
[08.04.2020 8:59:20.808 AM Diagnostic] 	C:\Workspaces\Test\Tests\bin\Debug\netcoreapp3.1\Assembly.Tests.dll
[08.04.2020 8:59:20.820 AM Informational] ---------- Discovery started ----------
[08.04.2020 8:59:20.823 AM Diagnostic] TelemetrySession: Creating the event: VS/UnitTest/TestWindow/Ext/RunSettingsService 
[08.04.2020 8:59:20.823 AM Diagnostic] Event:VS/UnitTest/TestWindow/Ext/RunSettingsService key: VS.UnitTest.TestWindow.RunSettingsService.Name value:VSTest Run Configuration
[08.04.2020 8:59:20.823 AM Diagnostic] TelemetrySession: Creating the event: VS/UnitTest/TestWindow/Ext/RunSettings 
[08.04.2020 8:59:20.823 AM Diagnostic] Event:VS/UnitTest/TestWindow/Ext/RunSettings key: VS.UnitTest.TestWindow.RunSettings.Services value:1
[08.04.2020 8:59:20.824 AM Informational] ========== Discovery skipped: All test containers are up to date ==========
[08.04.2020 8:59:20.824 AM Diagnostic] File timestamp remains 08.04.2020 08:45:43 for C:\Workspaces\Test\Tests\bin\Debug\netcoreapp3.1\Assembly.Tests.dll
[08.04.2020 8:59:20.826 AM Diagnostic] Project Assembly.Tests references test adapter: NUnit3TestAdapter, version 3.15.1
[08.04.2020 8:59:20.852 AM Informational] Executing test method: Tests.SomeClassTests.Sum_ReturnSum
[08.04.2020 8:59:20.902 AM Informational] ---------- Run started ----------
[08.04.2020 8:59:20.902 AM Diagnostic] TelemetrySession: Creating the event: VS/UnitTest/TestWindow/Ext/RunSettingsService 
[08.04.2020 8:59:20.903 AM Diagnostic] Event:VS/UnitTest/TestWindow/Ext/RunSettingsService key: VS.UnitTest.TestWindow.RunSettingsService.Name value:VSTest Run Configuration
[08.04.2020 8:59:20.903 AM Diagnostic] TelemetrySession: Creating the event: VS/UnitTest/TestWindow/Ext/RunSettings 
[08.04.2020 8:59:20.903 AM Diagnostic] Event:VS/UnitTest/TestWindow/Ext/RunSettings key: VS.UnitTest.TestWindow.RunSettings.Services value:1
[08.04.2020 8:59:20.904 AM Informational] ========== Run finished: 0 tests run (0:00:00,0013159) ==========`

Visual Studio in version 16.3.10
Main project and Test project in .NET Core 3.1

**Comment by mikkelbu** [comment]
@tomaszskrobek Does it work if you use the latest version of the test adapter (version 3.16.1)?

**Comment by tomaszskrobek** [comment]
Yes, I use latest version but I've found a funny thing... If I ran test from Test Explorer window or use mouse right click in file (somewhere in class, not in the test) and use Run Test(s) option the test is running correctly, but when I make a right click in test code and choose Run Test(s) option the test is not running. What could be a cause of that behavior?

**Comment by rprouse** [comment]
I'm going to move this issue over to the adapter repo. We have several similar issues that were introduced by Visual Studio when they added real time test discovery.
